--- openchange-openchange-2.3-VULCAN.orig/configure.ac	2024-04-09 11:55:51.678247965 +0200
+++ openchange-openchange-2.3-VULCAN/configure.ac	2024-04-09 20:15:46.847384577 +0200
@@ -464,6 +464,18 @@
 AC_CHECK_HEADER([samba/version.h],, AC_MSG_ERROR([Could not find Samba4 headers. Re-run ./configure with --with-samba=XXX where
  XXX is the prefix that Samba4 was installed to.]))
 
+AC_MSG_CHECKING(samba private libraries suffix)
+samba_private_libdir="$($PKG_CONFIG --variable=libdir samba-hostconfig)/samba"
+if test -e "$samba_private_libdir/libndr-samba-private-samba.so"; then
+    SMB_PRIV_LIB_SFX=private-samba
+elif test -e "$samba_private_libdir/libndr-samba-samba4.so"; then
+    SMB_PRIV_LIB_SFX=samba4
+else
+    AC_MSG_ERROR([failed])
+fi
+AC_MSG_RESULT($SMB_PRIV_LIB_SFX)
+AC_SUBST(SMB_PRIV_LIB_SFX)
+
 . `dirname $0`/script/samba4_ver.sh
 
 OC_CHECK_SAMBA_VERSION([$SAMBA4_RELEASE],[$SAMBA4_GIT_VER-GIT-$SAMBA4_GIT_REV])
--- openchange-openchange-2.3-VULCAN.orig/config.mk.in	2024-04-09 19:08:35.278698007 +0200
+++ openchange-openchange-2.3-VULCAN/config.mk.in	2024-04-09 18:58:37.035533217 +0200
@@ -68,7 +68,8 @@
 TALLOC_LIBS=@TALLOC_LIBS@
 
 CFLAGS+=$(SAMBA_CFLAGS) $(LDB_CFLAGS) $(TALLOC_CFLAGS) $(TDB_CFLAGS) $(THREAD_CFLAGS) $(NANOMSG_CFLAGS)
-LIBS+=$(SAMBA_LIBS) $(LDB_LIBS) $(TALLOC_LIBS) $(THREAD_LIBS)
+SMB_PRIV_LIB_SFX=@SMB_PRIV_LIB_SFX@
+LIBS+=$(SAMBA_LIBS) $(LDB_LIBS) $(TALLOC_LIBS) $(THREAD_LIBS) -ltevent-util -lsamba-errors -lndr-krb5pac ${libdir}/samba/libcli-ldap-common-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libcliauth-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libdcerpc-samba-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libldbsamba-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libndr-samba-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libsamba-debug-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libsamba-security-$(SMB_PRIV_LIB_SFX).so ${libdir}/samba/libsamdb-common-$(SMB_PRIV_LIB_SFX).so
 LDFLAGS+=@LDFLAGS@
 
 THREAD_LIBS=@THREAD_LIBS@
--- openchange-openchange-2.3-VULCAN/Makefile.orig	2015-05-16 17:22:04.000000000 +0200
+++ openchange-openchange-2.3-VULCAN/Makefile	2021-03-26 18:48:45.049437480 +0100
@@ -368,6 +368,9 @@
 	@echo "Linking $@"
 	@$(CXX) $(DSOOPT) $(CXX11FLAGS) $(CXXFLAGS) $(LDFLAGS) -Wl,-soname,libmapipp.$(SHLIBEXT).$(LIBMAPIPP_SO_VERSION) -o $@ $^ $(LIBS) 
 
+libmapipp.$(SHLIBEXT).$(LIBMAPI_SO_VERSION): libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION)
+	ln -fs $< $@
+
 libmapixx-installpc:
 	@echo "[*] install: libmapi++ pc files"
 	$(INSTALL) -d $(DESTDIR)$(libdir)/pkgconfig
@@ -858,8 +861,7 @@
 							mapiproxy/libmapiserver/libmapiserver_oxorule.po	\
 							mapiproxy/libmapiserver/libmapiserver_oxcperm.po	\
 							mapiproxy/libmapiserver/libmapiserver_oxcdata.po	\
-							ndr_mapi.po				\
-							gen_ndr/ndr_exchange.po
+							libmapi.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $(DSOOPT) $(LDFLAGS) -Wl,-soname,libmapiserver.$(SHLIBEXT).$(LIBMAPIPROXY_SO_VERSION) $^ $(LIBS)
 
@@ -1887,9 +1889,11 @@
 
 libqtmapi.$(SHLIBEXT).$(PACKAGE_VERSION): 	\
 	qt/lib/foldermodel.o			\
-	qt/lib/messagesmodel.o
+	qt/lib/messagesmodel.o                  \
+	libmapipp.$(SHLIBEXT).$(LIBMAPIPP_SO_VERSION) \
+	libmapi.$(SHLIBEXT).$(LIBMAPIPP_SO_VERSION)
 	@echo "Linking $@"
-	@$(CXX) $(DSOOPT) $(CXX11FLAGS) $(CXXFLAGS) $(LDFLAGS) -Wl,-soname,libqtmapi.$(SHLIBEXT).$(LIBQTMAPI_SO_VERSION) -o $@ $^ $(LIBS)
+	@$(CXX) $(DSOOPT) $(CXX11FLAGS) $(CXXFLAGS) $(LDFLAGS) -Wl,-soname,libqtmapi.$(SHLIBEXT).$(LIBQTMAPI_SO_VERSION) -o $@ $^ -lQtGui -lQtCore $(LIBS)
 
 
 qt/demo/demoapp: qt/demo/demoapp.o 				\
